docs/modules/ROOT/pages/integrations/database.adoc

```diff
@@ -0,0 +1,114 @@
+= Database Integrations
+
+== Overview
+
+KubeArchive requires a database to store resources.
+This document list the databases supported, how to set up the database schema
+and how to add new database integrations.
+
+=== PostgreSQL
+
+The only currently full supported.
+KubeArchive integration test suite runs with PostgreSQL 16.6
+
+The PostgreSQL implementation is available
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/postgresql.go[here].
+
+=== MariaDB
+
+This integration is still in progress.
+
+The MariaDB implementation is available
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/mariadb.go[here].
+
+== Configuration and Customization
+
+KubeArchive offers a [filename]`kubearchive.sql` file with the DDL to create
+the schema as part of the installation.
+
+[IMPORTANT]
+====
+The provided schema includes indexes.
+
+KubeArchive users should review and customize the indexes
+if needed based on the expected queries and the amount and
+the distribution of the expected archived data.
+====
+
+Specific instructions for the configuration of the PostgreSQL Database are available in
+xref:/getting-started/instllation.adoc#_install_and_configure_kubearchive[Install and Configure KubeArchive].
+
+== Add new database integrations
+
+To add a new database integration follow the instructions:
+
+1. Create a new file under `pkg/database` with the name of the new database type.
+2. Include an
+link:https://go.dev/doc/effective_go#init[init function]
+that:
+
+* Inserts in the
+link:https://github.com/maruiz93/kubearchive/blob/postgresql-version-update/pkg/database/database.go#L27[`RegisteredDBCreators`]
+map an entry keyed with the name of the DB type with a
+link:https://github.com/maruiz93/kubearchive/blob/postgresql-version-update/pkg/database/database.go#L24[`newDBCreatorFunc`]
+implementation as value.
+
+* Inserts in the
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/database.go#L26[`RegisteredDatabases`]
+map an entry keyed with the name of the DB type with a
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/database.go#L23[`newDatabaseFunc`]
+as value.
+
+=== The `newDBCreatorFunc`
+
+This function receives the set of environment variables with the database configuration and returns a
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/creator.go[`DBCreator`]
+implementation.
+
+The recommendation is that the `DBCreator` implementation include the `env` param as parameter
+so the `GetDriverName` and the `GetConnectionString` can access them.
```

Comment:
```suggestion
The `DBCreator` implementation must include the `env` param as parameter
so the `GetDriverName` and the `GetConnectionString` can access them.
```

Reply:
I've replaced it with a `should`.

You can potentially do a working `GetConnectionString` method that doesn't depend on an `env`, but you shouldn't do that because that breaks our "development guidelines".

Reply:
I'm fine with should instead of must. However I think this should not be about what you could do, but about what we want you to do as maintainers of the tool.